prime-router/src/main/kotlin/azure/EmailSenderFunction.kt

```diff
@@ -0,0 +1,140 @@
+    data class SenderTosFormJSON(
+        val title: String,
+        val firstName: String,
+        val lastName: String,
+        val email: String,
+        val territory: String,
+        val organizationName: String,
+        val operatesInMultipleStates: Boolean,
+        val agreedToTermsOfService: Boolean
+    )
+}
+
+class EmailSenderFunction {
+
+    @FunctionName("emailRegisteredOrganization")
+    @StorageAccount("AzureWebJobsStorage")
+    fun emailRegisteredOrganization(
+        @HttpTrigger(
+            name = "emailRegisteredOrganization",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "email-registered"
+        )
+        request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        val logger: Logger = context.logger
+        var ret = request.createResponseBuilder(HttpStatus.BAD_REQUEST)
+
+        if (request.body !== null) {
+            logger.info(request.body)
+            ret.status(sendRegistrationConfirmation(request.body!!, logger))
+        }
+
+        return ret.build()
+    }
+
+    private fun sendRegistrationConfirmation(requestBody: String, logger: Logger): HttpStatus {
+        val gson = Gson()
+        val body = gson.fromJson<SenderTosRequest.SenderTosFormJSON>(
+            requestBody,
+            SenderTosRequest.SenderTosFormJSON::class.java
+        )
+        val title = body.title
+        val firstName = body.firstName
+        val lastName = body.lastName
+        val email = body.email
+        val territory = body.territory
+        val organizationName = body.organizationName
+        val operatesInMultipleStates = body.operatesInMultipleStates
+        val agreedToTermsOfService = body.agreedToTermsOfService
+
+        var status: HttpStatus = HttpStatus.NOT_FOUND
+        val content: Content = Content()
+        content.type = "plain/text"
+
+        val p: Personalization = Personalization()
+        p.addTo(Email(REGISTER_EMAIL))
+        p.addCc(Email(email))
+        p.addDynamicTemplateData(
+            "formData",
+            object {
+                var title = title
+                var firstName = firstName
+                var lastName = lastName
+                var email = email
+                var territory = territory
+                var organizationName = organizationName
+                var operatesInMultipleStates = operatesInMultipleStates
+                var agreedToTermsOfService = agreedToTermsOfService
+            }
+        )
+
+        val mail: Mail = Mail()
+        mail.setTemplateId(TEMPLATE_ID)
+        mail.addPersonalization(p)
+        mail.setSubject(REGISTER_SUBJECT_BASE + organizationName)
+        mail.setFrom(Email(NO_REPLY_EMAIL))
+
+        var sendgridId: String? = SecretHelper.getSecretService().fetchSecret("SENDGRID_ID")
+        var response: Response = Response()
+
+        if (sendgridId !== null) {
```

Comment:
Maybe log an error if it comes back null or empty to help with troubleshooting?

Reply:
+1 We have a simple info log down below in the else but a more comprehensive error would be nice